src/main/java/uk/gov/hmcts/reform/civil/service/PaymentRequestUpdateCallbackService.java

```diff
@@ -3,167 +3,236 @@
+        return hearingPayment == null || hearingPayment.getStatus() == PaymentStatus.FAILED;
+    }
 
-        CaseDataContent caseDataContent = buildCaseDataContent(
-            startEventResponse,
-            caseData
+    private boolean isClaimIssuePaymentFailed(CaseData caseData) {
+        PaymentDetails claimIssuePayment = caseData.getClaimIssuedPaymentDetails();
+        return claimIssuePayment == null || claimIssuePayment.getStatus() == PaymentStatus.FAILED;
+    }
+
+    @Retryable(value = CaseDataUpdateException.class, backoff = @Backoff(delay = 500))
+    public void updatePaymentStatus(FeeType feeType, String caseReference, CardPaymentStatusResponse paymentStatusResponse) {
+        try {
+            Long caseId = Long.valueOf(caseReference);
+            CaseDetails caseDetails = coreCaseDataService.getCase(caseId);
+            CaseData caseData = caseDetailsConverter.toCaseData(caseDetails);
+            caseData = updateCaseDataWithPaymentDetails(paymentStatusResponse, caseData, feeType);
+
+            createAndSubmitEvent(caseData, caseId, feeType, false);
+        } catch (Exception ex) {
+            log.error("Error updating payment status for case {}: {}", caseReference, ex.getMessage());
+            throw new CaseDataUpdateException();
+        }
+    }
+
+    private void handlePaymentUpdate(ServiceRequestUpdateDto dto, CaseData caseData, FeeType feeType, Long caseId, boolean isValidPayment) {
+        caseData = updateCaseDataWithPaymentDetails(dto, caseData, feeType);
+        if (isValidPayment) {
+            CardPaymentStatusResponse paymentStatusResponse = buildPaymentStatusResponse(dto);
+            updatePaymentStatus(feeType, dto.getCcdCaseNumber(), paymentStatusResponse);
+        }
+        createAndSubmitEvent(caseData, caseId, feeType, true);
+    }
+
+    private CardPaymentStatusResponse buildPaymentStatusResponse(ServiceRequestUpdateDto dto) {
+        return CardPaymentStatusResponse.builder()
+            .paymentReference(dto.getPayment().getPaymentReference())
+            .status(PaymentStatus.SUCCESS.name())
+            .build();
+    }
+
+    private void createAndSubmitEvent(CaseData caseData, Long caseId, FeeType feeType, boolean isCallback) {
+        CaseEvent event = determineEvent(caseData, feeType, isCallback);
+        StartEventResponse startEvent = coreCaseDataService.startUpdate(
+            String.valueOf(caseId),
+            event
         );
 
-        coreCaseDataService.submitUpdate(caseId, caseDataContent);
+        CaseDataContent caseDataContent = CaseDataContent.builder()
+            .eventToken(startEvent.getToken())
+            .event(Event.builder().id(startEvent.getEventId()).build())
+            .data(caseData.toMap(objectMapper))
+            .build();
+
+        coreCaseDataService.submitUpdate(String.valueOf(caseId), caseDataContent);
+    }
+
+    private CaseEvent determineEvent(CaseData caseData, FeeType feeType, boolean isCallback) {
```

Comment:
Same here re booleans - don't really understand what the condition is so makes it hard to understand the business context. I'd just move the logic into this method and think about a name that's a bit more descriptive